quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/KeycloakMain.java

```diff
@@ -164,17 +170,41 @@ public int run(String... args) throws Exception {
     private void createAdminUser() {
         String adminUserName = System.getenv(KEYCLOAK_ADMIN_ENV_VAR);
         String adminPassword = System.getenv(KEYCLOAK_ADMIN_PASSWORD_ENV_VAR);
+        String tmpFirstName = System.getenv(KEYCLOAK_ADMIN_FIRSTNAME_ENV_VAR);
+        String tmpLastName = System.getenv(KEYCLOAK_ADMIN_LASTNAME_ENV_VAR);
+        String tmpEmail = System.getenv(KEYCLOAK_ADMIN_EMAIL_ENV_VAR);
 
-        if ((adminUserName == null || adminUserName.trim().length() == 0)
-                || (adminPassword == null || adminPassword.trim().length() == 0)) {
+        if (StringUtil.isBlank(adminUserName) || StringUtil.isBlank(adminPassword)) {
             return;
         }
 
+        // try to create admin user only with username and password
+        if (StringUtil.isBlank(tmpFirstName)) {
+            tmpFirstName = adminUserName;
+        }
+
+        if (StringUtil.isBlank(tmpLastName)) {
+            tmpLastName = adminUserName;
+        }
+
+        if (StringUtil.isBlank(tmpEmail)) {
+            tmpEmail = adminUserName + "@" + KEYCLOAK_ADMIN_DEFAULT_EMAIL_DOMAIN;
```

Comment:
Probably a corner-case, but in case that someone uses `KEYCLOAK_ADMIN=admin@some_domain.com` and `KEYCLOAK_ADMIN_PASSWORD=whatever` and not other variables, then tmpEmail would be auto-generated to `admin@some_domain.com@keycloak.test` , which is not valid email AFAIK and hence it would fail in the next check without any warning displayed to the administrator. I suppose that for this case when `adminUserName` is already a valid email, we might want to either:
- Use the `tmpEmail` just based on the `adminUserName`
- Skip the admin user creation like you did, but in this case, it would be good to write some WARNING or ERROR to the server.log to add a hint to admin user why the user creation failed? IMO writing the WARNING/ERROR to the server.log would be good regardless, so administrator knows why admin user creation failed.